src/execution_plan/ops/op_semi_apply.c

```diff
@@ -88,16 +90,23 @@ Record AntiSemiApplyConsume(OpBase *opBase) {
 		op->r = OpBase_Consume(op->bound_branch);
 		if(!op->r) return NULL; // Depleted.
 
+		// Propagate record to the top of the Match stream.
+		// (Must clone the Record, as it will be freed in the Match stream.)
+		if(op->op_arg) Argument_AddRecord(op->op_arg, OpBase_CloneRecord(op->r));
 		/* Try pulling right stream
 		 * return bounded stream record if match stream returns NULL. */
-		Record righthand_record = _pullFromMatchStream(op);
-		if(righthand_record) {
+		bool matched = false;
+		Record rhs_record;
+		// Retrieve Records from the Match stream until it's depleted.
+		while((rhs_record = _pullFromMatchStream(op))) {
 			/* managed to get a record from match stream,
 			 * free it and pull again from left handside. */
-			OpBase_DeleteRecord(righthand_record);
+			OpBase_DeleteRecord(rhs_record);
 			OpBase_DeleteRecord(op->r);
 			op->r = NULL;
```

Comment:
If you want to completely deplete rhs, please move 
```
OpBase_DeleteRecord(rhs_record);
OpBase_DeleteRecord(op->r);
```
to a different scope after the while iteration

Reply:
Moving `OpBase_DeleteRecord(op->r)` outside, but I think that `OpBase_DeleteRecord(rhs_record)` belongs in this loop (as the condition generates `rhs_record`, but we'll never wish to access it).